### PR TITLE
Small fixes in telemetry and plugin init

### LINKED ISSUE #103: [BUG] - Preevy should print meaningful error if docker executable not found
Currently, preevy print this error:
```
Error: spawn docker ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at processTicksAndRejections (node:internal/process/task_queues:82:2
```
It's very aggressive and we should print a friendlier message

### LINKED ISSUE #114: [BUG] - Preevy down require docker-compose context to work
Using `preevy down --id="myenv"` should be able to destroy the environment in similar fashion to docker-compose -p

### LINKED ISSUE #115: [BUG] - Preevy should not fail on 'init' when Docker is not installed
Currently, loading plugins is dependent on docker-compose and Preevy crashes when there's no docker client.
Ideally, we should be able to run without plugins most command (maybe add warning), and the behavior should be handled more gracefully.

### PR DESCRIPTION
**Pull request by Yshayy**
- closes #114 
- closes #115
- closes #103 
In addition, this fix a bug of unhandled exception and additional telemetry

### CODE REVIEW

File: packages/cli/src/commands/down.ts [hunk 1]
```diff
@@ -28,11 +28,11 @@ export default class Down extends DriverCommand<typeof Down> {
     const log = this.logger
     const { flags } = await this.parse(Down)
     const driver = await this.driver()
-    const userModel = await this.ensureUserModel()
-    const projectName = userModel.name
-    log.debug(`project: ${projectName}`)
-    const envId = flags.id ?? await findAmbientEnvId(projectName)
-
+    const [envId, userModel] = flags.id ? [flags.id, { name: flags.id }] : await (async () => {
```
Comment (royra) [inline packages/cli/src/commands/down.ts:31]: `envId` is incorrect here - missing branch suffix.
what is this fixing?

File: packages/cli/src/hooks/init/load-plugins.ts [hunk 2]
```diff
@@ -1,12 +1,19 @@
 import { Hook as OclifHook } from '@oclif/core'
 import { initHook } from '@preevy/cli-common'
+import { telemetryEmitter } from '@preevy/core'
 
 const wrappedHook: OclifHook<'init'> = async function wrappedHook(...args) {
   try {
+    const { id } = args[0]
```
Comment (royra) [inline packages/cli/src/hooks/init/load-plugins.ts:7]: IMO this function shouldn't include any logic besides wrapping with try..catch, you can move the id check to the hook itself.

File: packages/core/src/child-process.ts [hunk 3]
```diff
@@ -71,7 +72,7 @@ export const childProcessStdoutPromise = async (
   return orderedOutput(output).stdout().toString('utf-8').trim()
 }
 
-export const spawnPromise = (
+export const spawnPromise = async (
```
Comment (royra) [inline packages/core/src/child-process.ts:75]: redundant `async`

File: packages/core/src/compose/client.ts [hunk 4]
```diff
@@ -35,21 +41,26 @@ const composeClient = (
   executer: Executer,
   composeFiles: string[] | Buffer,
 ) => {
-  const execComposeCommand = (args: string[]) => executer({
+  const execComposeCommand = async (args: string[]) => executer({
```
Comment (royra) [inline packages/core/src/compose/client.ts:44]: redundant `async`

File: packages/core/src/compose/client.ts [hunk 5]
```diff
@@ -116,11 +127,11 @@ export const localComposeClient = (
     ...args: Parameters<typeof spawnComposeArgs>
   ) => addStdIn(spawn(...spawnComposeArgs(...args)))
 
-  const spawnComposePromise = (
+  const spawnComposePromise = async (
```
Comment (royra) [inline packages/core/src/compose/client.ts:130]: redundant `async`

File: packages/core/src/compose/client.ts [hunk 6]
```diff
@@ -116,11 +127,11 @@ export const localComposeClient = (
...
     ...args: Parameters<typeof spawnComposeArgs>
   ) => childProcessPromise(spawnCompose(...args))
 
-  const executer: Executer = ({ args }) => childProcessStdoutPromise(spawnCompose(args, {}))
+  const executer: Executer = async ({ args }) => childProcessStdoutPromise(spawnCompose(args, {}))
```
Comment (royra) [inline packages/core/src/compose/client.ts:134]: redundant `async`

File: packages/core/src/telemetry/machine-id.ts [hunk 7]
```diff
@@ -33,12 +34,13 @@ const calcMachineId = async () => {
 }
 
 export const memoizedMachineId = async (dataDir: string) => {
+  const dir = localFs(dataDir)
```
Comment (royra) [inline packages/core/src/telemetry/machine-id.ts:37]: why is `localFs` needed here?

**Review by royra — COMMENTED** [review]
LGTM, see minor comments

**Review by royra — APPROVED** [review]

File: packages/cli/src/commands/down.ts (hunk 1 above)
Comment (Yshayy) [reply packages/cli/src/commands/down.ts:31]: The case of using down --id without docker/compose. (#114)

Comment (royra) [reply packages/cli/src/commands/down.ts:31]: if the project name is only needed for the `debug.log` call, I suggest removing it to simplify:
```suggestion
    const envId = flags.id ?? await findAmbientEnvId((await this.ensureUserModel()).name)
```

File: packages/core/src/child-process.ts (hunk 3 above)
Comment (Yshayy) [reply packages/core/src/child-process.ts:75]: The async isn't redundant, if `spawn` logic throw the error is not wrapped.
Ideally, any function that return a Promise should give a strong guarantee that the exception is always wrapped in the promise return value. 
`async` force that, for example:
```
await ((function(){ throw "abc"; return Promise.resolve(5)}()).catch(e=> { }))
```
throws "abc"

```
await ((async function(){ throw "abc"; return Promise.resolve(5)}()).catch(e=> { }))
```
doesn't throw

File: packages/core/src/telemetry/machine-id.ts (hunk 7 above)
Comment (Yshayy) [reply packages/core/src/telemetry/machine-id.ts:37]: in container environments, datadir folder didn't exist

File: packages/cli/src/commands/down.ts (hunk 1 above)
Comment (Yshayy) [reply packages/cli/src/commands/down.ts:31]: That was the variable before.
```const envId = flags.id ?? await findAmbientEnvId(projectName)```

**Comment by Yshayy** [comment]
Went with `standard` approach:
https://github.com/standard/standard/issues/1442
https://github.com/standard/eslint-config-standard-with-typescript/pull/253

Remote no-return-await (to avoid synchronous throws and use async stack-traces: https://v8.dev/blog/fast-async#improved-developer-experience) and force typescript to require the use of `await` when using async

**Pull request merged**